nrf_wifi/fw_if/umac_if/src/tx.c

```diff
@@ -737,6 +918,53 @@ enum nrf_wifi_status tx_cmd_prepare(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx,
 	return NRF_WIFI_STATUS_FAIL;
 }
 
+#ifdef CONFIG_NRF700X_RAW_DATA_TX
+enum nrf_wifi_status rawtx_cmd_init(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx,
+				    void *txq,
+				    int desc,
+				    int peer_id)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	struct host_rpu_msg *umac_cmd = NULL;
+	unsigned int len = 0;
+	struct nrf_wifi_fmac_dev_ctx_def *def_dev_ctx = NULL;
+
+	def_dev_ctx = wifi_dev_priv(fmac_dev_ctx);
+
+	len += sizeof(struct nrf_wifi_cmd_raw_tx);
+
+	len *= nrf_wifi_utils_list_len(fmac_dev_ctx->fpriv->opriv, txq);
+
+	umac_cmd = umac_cmd_alloc(fmac_dev_ctx,
+				  NRF_WIFI_HOST_RPU_MSG_TYPE_SYSTEM,
+				  len);
+
+	status = rawtx_cmd_prepare(fmac_dev_ctx,
+				   umac_cmd,
+				   desc,
+				   txq,
+				   peer_id);
+
+	if (status != NRF_WIFI_STATUS_SUCCESS) {
+		nrf_wifi_osal_log_err(fmac_dev_ctx->fpriv->opriv,
+				      "%s: rawtx_cmd_prepare failed",
+				      __func__);
+
```

Comment:
Don't we need to free the `umac_cmd` here?

Reply:
no - umac_cmd will get freed later once the command is been processed.